### PR TITLE
CO-657 Store file data version wise in object and pass it directly as per server request.

### PR DESCRIPTION
**Pull request by ab14bhardwaj**
### What do you want to achieve?
-> Store file data version wise in object and pass it directly as per server request.

### How was the code tested?

-> By running script for both v1 and v2 versions.

### What new thing you came across while writing this code? 
-> [util.promisify()](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original)

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/controller.js [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
-exports.getPlugin = (req, res) => {
-  //var MCK_CONTEXTPATH =req.protocol+"://"+req.headers.host;
+exports.getPlugin = async (req, res) => {
+  const MCK_PLUGIN_VERSION = req.params.version;
+  if (!MCK_PLUGIN_VERSION) {
+    res.send("err while getting plugin...");
+    return console.log("unable to get plugin version");
+  }
+  var data = Object.keys(pluginVersionData).length ? pluginVersionData[MCK_PLUGIN_VERSION] : await generatePluginFile(req, res);
+  res.setHeader('content-type', 'application/javascript');
```
Comment (devashishmamgain) [inline webplugin/controller.js]: @ab14bhardwaj can this case impact anything on any browser? standard is "Content-Type". 
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields
headers always starts with Capital letter and have - as splitter.

File: webplugin/controller.js [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-      .replace(":MCK_STATICPATH", MCK_STATICPATH).replace(":PRODUCT_ID", "kommunicate");
-    res.setHeader('content-type', 'application/javascript');
-    res.send(plugin);
-    console.log("plugin code sent successfully");
-  });
-}
+    .replace(":MCK_THIRD_PARTY_INTEGRATION", JSON.stringify(MCK_THIRD_PARTY_INTEGRATION))
+    .replace(":MCK_STATICPATH", MCK_STATICPATH).replace(":PRODUCT_ID", "kommunicate")
+    .replace(":MCK_PLUGIN_VERSION", MCK_PLUGIN_VERSION).replace(":PLUGIN_SETTINGS", JSON.stringify(PLUGIN_SETTING))
+    .replace(":MCK_STATICPATH", MCK_STATICPATH).replace(":PRODUCT_ID", "kommunicate");
```
Comment (devashishmamgain) [inline webplugin/controller.js]: @ab14bhardwaj is PRODUCT_ID getting replace twice? looks like old code had same problem.

File: webplugin/controller.js (hunk 1 above)
Comment (ab14bhardwaj) [reply webplugin/controller.js]: @devashishmamgain Changes done.

File: webplugin/controller.js (hunk 2 above)
Comment (ab14bhardwaj) [reply webplugin/controller.js]: @devashishmamgain Changes done.

**Comment by ab14bhardwaj** [comment]
@devashishmamgain  @Suraj-Kumar @devashishmamgain Please don't merge the pull request as it would be tested from this branch only.
@vipinsonu1 Please do the testing for https://applozic.atlassian.net/browse/CO-657 from this branch.

**Pull request merged**